File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        HashMap<Integer,Integer> map = new HashMap<>();
        for(int i: nums){
            map.put(i,map.getOrDefault(i,0)+1);
        }
        int result[] = {0,0};
        for(Map.Entry<Integer,Integer> mapValues: map.entrySet()){
            if(result[1]<mapValues.getValue()){
                result[0] = mapValues.getKey();
                result[1] = mapValues.getValue();
            }
        }
        return result[0];
    }
}

File: 0169-majority-element/0169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        mapDictionary = defaultdict(int)
        result = [0,0]
        for i in nums:
            mapDictionary[i] +=1
            if mapDictionary[i] > result[1]:
                result[0] =i
                result[1] = mapDictionary[i]
        return result[0]